### PR TITLE
Clean up some players references

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

**Review by ephphatha — CHANGES_REQUESTED** [review]

File: Source/items.cpp [hunk 1]
```diff
@@ -3769,75 +3762,59 @@ void GetItemStr(ItemStruct &item)
...
-
 	if (cii >= NUM_INVLOC)
-		pi = &Players[pnum].InvList[cii - NUM_INVLOC];
-	else
-		pi = &Players[pnum].InvBody[cii];
+		cii -= NUM_INVLOC;
 
-	pi->_iIdentified = true;
-	CalcPlrInv(pnum, true);
+	player.InvBody[cii]._iIdentified = true;
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/items.cpp:3770]: wow i compleatly overlooked that it was two different arrays, thanks for catching

File: Source/items.h [hunk 2]
```diff
@@ -409,11 +411,14 @@ bool IsItemAvailable(int i);
...
-void CalcPlrInv(int p, bool Loadgfx);
+void CalcPlrItemVals(PlayerStruct &player, bool Loadgfx);
+void CalcPlrInv(PlayerStruct &player, bool Loadgfx);
 void SetPlrHandItem(ItemStruct &item, int itemData);
 void GetPlrHandSeed(ItemStruct *h);
-void GetGoldSeed(int pnum, ItemStruct *h);
+/**
+ * @brief Set a new unique seed value on the given item
+ */
+void GetGoldSeed(PlayerStruct &player, ItemStruct &gold);
```
Comment (obligaron) [inline Source/items.h:421]: The name is misleading. Shouldn't it be `SetGoldSeed`?

```suggestion
/**
 * @brief Set a new unique seed value on the given item
 */
void SetGoldSeed(PlayerStruct &player, ItemStruct &gold);
```

File: Source/items.cpp [hunk 3]
```diff
@@ -28,6 +28,7 @@
 #include "utils/language.h"
 #include "utils/math.h"
 #include "utils/stdcompat/algorithm.hpp"
+#include "player.h"
```
Comment (obligaron) [inline Source/items.cpp:31]: nit: include at alphabetical order?

File: Source/items.cpp [hunk 4]
```diff
@@ -3769,75 +3762,59 @@ void GetItemStr(ItemStruct &item)
...
 
-	if (pnum == MyPlayerId)
+	if (&player == &Players[MyPlayerId])
 		NewCursor(CURSOR_HAND);
 }
 
-void DoOil(int pnum, int cii)
+void DoOil(PlayerStruct &player, int cii)
 {
 	if (cii < NUM_INVLOC && cii != INVLOC_HEAD && (cii <= INVLOC_AMULET || cii > INVLOC_CHEST))
```
[the remark this answers is not in the document]
Comment (obligaron) [reply Source/items.cpp:3811]: I think this check is entirely not needed. 🙁 
Cause you are also not allowed to apply oil to rings or amulets in inventory.
This check is performed in `ApplyOilToItem` and is also executed when oil is applied to equipped jewellery.

**Review by obligaron — COMMENTED** [review]
Same issues at @ephphatha and two nits, else lgtm. 😉

File: Source/items.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/items.cpp:31]: I tried :(

Comment (obligaron) [reply Source/items.cpp:31]: Don't give up! We believe in you!

File: Source/items.cpp [hunk 5]
```diff
@@ -3769,75 +3762,59 @@ void GetItemStr(ItemStruct &item)
...
 		return;
-	auto &player = Players[pnum];
 	if (!ApplyOilToItem(player.InvBody[cii], player))
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/items.cpp:3813]: Probably only works cause InvBody and InvList are next to each other

**Review by obligaron — APPROVED** [review]

File: Source/items.cpp [hunk 6]
```diff
@@ -3769,28 +3762,26 @@ void GetItemStr(ItemStruct &item)
...
+void CheckIdentify(PlayerStruct &player, int cii)
 {
 	ItemStruct *pi;
 
 	if (cii >= NUM_INVLOC)
-		pi = &Players[pnum].InvList[cii - NUM_INVLOC];
+		pi = &player.InvList[cii - NUM_INVLOC];
 	else
-		pi = &Players[pnum].InvBody[cii];
+		pi = &player.InvBody[cii];
```
Comment (ephphatha) [inline Source/items.cpp:3772]: Nit: Other functions with this code use braces, inconsistent with the style guide for single line cases.

However given you're touching the lines anyway extracting this to a function returning an `ItemStruct&` would be great. Then we know all four functions are using the same logic and the initialisation can also be made simpler.

**Review by ephphatha — APPROVED** [review]
No logic issues that I could see, one nit if you felt like it :D

File: Source/items.cpp (hunk 6 above)
Comment (AJenbo) [reply Source/items.cpp:3772]: Saw it, but decided i needed to move on and leave something for someone else :)

**Pull request merged**